plugin/src/main/java/com/denizenscript/denizen/nms/interfaces/WorldHelper.java

```diff
@@ -15,4 +15,22 @@ public interface WorldHelper {
     default Location getNearestBiomeLocation(Location start, BiomeTag biome) {
         throw new UnsupportedOperationException();
     }
+
+    boolean areEnoughSleeping(World world, int percentage);
+
+    boolean areEnoughDeepSleeping(World world, int percentage);
+
+    int getSkyDarken(World world);
+
+    boolean isDay(World world);
+
+    boolean isNight(World world);
+
+    // for setting the time without firing a CUSTOM TimeSkipEvent
+    void setDayTime(World world, long time);
+
+    void wakeUpAllPlayers(World world);
+
+    // for clearing weather without ignoring possible raised event results
+    void clearWeather(World world);

```

Comment:
Isn't there API for that? iirc it's
`setRainTime` -> `World#setWeatherDuration`
`setThunderTime` -> `World#setThunderDuration`
`setRaining` -> `World#setStorm`
`setThundering` -> `World#setThundering`

Reply:
The problem is those all fire events, which is not a vanilla behavior for the use case this method applies to

Reply:
Got that mixed up, the problem with this is that `setStorm` and `setThundering` indiscriminately set the durations to 0 regardless of the events they *do* fire. Could be worked around in theory by grabbing the durations before and re-setting after, but since this stuff already touches other internals I just went the extra mile for semi-cleanliness